src/goose/batch.clj

```diff
@@ -1,22 +1,26 @@
 (ns goose.batch
   (:require
+    [goose.defaults :as d]
     [goose.utils :as u]))
 
 (def status-in-progress "in-progress")
 (def status-complete "complete")
 
 (defn new
-  [{:keys [callback-fn-sym]} jobs]
+  [{:keys [queue retry-opts]}
+   {:keys [callback-fn-sym]}
+   jobs]
   (let [id (str (random-uuid))]
     {:id              id
      :callback-fn-sym callback-fn-sym
+     :queue           queue
+     :ready-queue     (d/prefix-queue queue)
+     :retry-opts      retry-opts
```

Comment:
The retry-queue in retry-opts isn't prefixed.  Line 47-48 in `goose.client` needs to be extracted to a function that all perform-* functions use before working on opts. 

I'm afraid this duplication of logic will be difficult to keep track in future. So, lets refactor now.

Reply:
Maybe a function like
```clojure
(defn- enhance-opts-for-internal-use
  [{:keys [queue retry-opts] :as opts}]
  (assoc opts
    :ready-queue (d/prefix-queue queue)
    :retry-opts (retry/prefix-queue-if-present retry-opts)))

(defn- create-job
  [{:keys [queue retry-opts ready-queue]} execute-fn-sym args]
  (j/new execute-fn-sym args queue ready-queue retry-opts))

(defn- enqueue
  ...
  (let [enhanced-opts (enhance-opts-for-internal-use opts)
        job (create-job enhanced-opts execute-fn-sym args)]
  ...)))

(defn perform-batch
   ...
   (let [enhanced-opts (enhance-opts-for-internal-use opts)
         jobs (map #(create-job enhanced-opts execute-fn-sym %) args-coll)
   ...)))

```

Reply:
Maybe before `goose.worker` calls `b/start-worker`, we should enhance the opts to have ready-queue. In that case, the function might go to utils. (This isn't necessary right now, I was just checking where all we're creating a `ready-queue` & saw workers doing it too so suggested).

Naming is tricky because this function is doing both adding a ready-queue & prefixing retry-queue. I'll leave that upto you.

Reply:
I've let the name be `enhance-opts-for-internal-use` and made this change in `goose.client` for now. 

We can perhaps move it to utils later? This involve making updates at more places, making the scope of this change larger, hence not picking it now.